### ISSUE TITLE
Staticcheck can't load some nested files in my project

### ISSUE THREAD

**Issue by marek2901** [dominikh/go-tools#567]
➜ ~ staticcheck -version
staticcheck (no version)

➜ ~ staticcheck -debug.version
staticcheck (no version)

Compiled with Go version: go1.12.8
Built without Go modules

➜ ~ go version
go version go1.12.8 darwin/amd64

➜ ~ go env
```
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/mark.dziewit/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/mark.dziewit/golang"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/opt/go/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/opt/go/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/mark.dziewit/marek2901/projects/project/src/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/qq/n_wbwgnn347c59kfxdq2ydk80000gn/T/go-build781349985=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

Command ran:
staticcheck ./...

Output:
-: could not load export data: no export data for "github.com/marek2901/project/src/common/ops/integrationtest" (compile)
-: could not load export data: no export data for "github.com/marek2901/project/src/someapi/integrationtest" (compile)
-: could not load export data: no export data for "github.com/marek2901/project/src/internalapi/integrationtest" (compile)

repo link private and confidential

**Comment by marek2901** [comment]
After cleaning and reinstalling (recompiling). It seems to work for the first time, then it fails constantly.

**Comment by dominikh** [comment]
I haven't seen this before. Is there anything special about your setup? Is your home directory on a network share? Is some process deleting files from /Users/mark.dziewit/Library/Caches/go-build? Can you reproduce the issue with any other packages, or just these particular ones?

**Comment by mvdan** [comment]
I ran into this while using staticcheck master at work. It seems like the issue involves a nested module and a package with tests. Here's a repro on staticcheck v0.0.0-20191003165122-a2fc4037bf9f and `go version devel +b106105e52 Sat Oct 5 08:29:48 2019 +0000 linux/amd64`:

```
mkdir foo; cd foo
go mod init test.tld/foo
mkdir bar; cd bar
go mod init test.tld/foo/bar
echo 'package bar' >bar_test.go
go test
staticcheck .
```

And my output:

```
$ mkdir foo; cd foo
$ go mod init test.tld/foo
go: creating new go.mod: module test.tld/foo
$ mkdir bar; cd bar
$ go mod init test.tld/foo/bar
go: creating new go.mod: module test.tld/foo/bar
$ echo 'package bar' >bar_test.go
$ go test
testing: warning: no tests to run
PASS
ok      test.tld/foo/bar        0.001s
$ staticcheck .
-: could not load export data: no export data for "test.tld/foo/bar" (compile)
```

Either module could potentially contain the package, which is what I think is causing the software to get confused. On the other hand, neither module requires the other, so there shouldn't be any ambiguity. staticcheck v0.0.1-2019.2.3 doesn't have this bug.

**Comment by dominikh** [comment]
This was fixed by 82a1013350519505150bbec002cf52f829dee7a9
